From f0909391b264d01794466956f3672afbe1a7d7b5 Mon Sep 17 00:00:00 2001
From: Marc-Aurel Zent <mzent@codeweavers.com>
Date: Thu, 30 Oct 2025 22:02:27 +0100
Subject: [PATCH] ntdll: Add is_pseudo_handle() helper.

Also fixes an off-by-one error in the pseudo-handle check.
---
 dlls/ntdll/unix/sync.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/dlls/ntdll/unix/sync.c b/dlls/ntdll/unix/sync.c
index 07ffe854104..ef6dfe1e2c1 100644
--- a/dlls/ntdll/unix/sync.c
+++ b/dlls/ntdll/unix/sync.c
@@ -556,6 +556,11 @@ static inline unsigned int inproc_sync_handle_to_index( HANDLE handle, unsigned
     return idx % INPROC_SYNC_CACHE_BLOCK_SIZE;
 }
 
+static inline BOOL is_pseudo_handle( HANDLE handle )
+{
+    return ((ULONG)(ULONG_PTR)handle >= 0xfffffffa);
+}
+
 static struct inproc_sync *cache_inproc_sync( HANDLE handle, struct inproc_sync *sync )
 {
     unsigned int entry, idx = inproc_sync_handle_to_index( handle, &entry );
@@ -563,7 +568,7 @@ static struct inproc_sync *cache_inproc_sync( HANDLE handle, struct inproc_sync
     int refcount;
 
     /* don't cache pseudo-handles; waiting on them is pointless anyway */
-    if ((ULONG)(ULONG_PTR)handle > 0xfffffffa) return sync;
+    if (is_pseudo_handle( handle )) return sync;
 
     if (entry >= INPROC_SYNC_CACHE_ENTRIES)
     {
-- 
GitLab

